File: tkpdstream/src/main/java/com/tokopedia/tkpdstream/chatroom/view/viewmodel/chatroom/SprintSaleAnnouncementViewModel.java
package com.tokopedia.tkpdstream.chatroom.view.viewmodel.chatroom;

import com.tokopedia.abstraction.base.view.adapter.Visitable;
import com.tokopedia.tkpdstream.chatroom.view.adapter.chatroom.typefactory.GroupChatTypeFactory;

import java.util.ArrayList;

/**
 * @author by nisie on 3/22/18.
 */

public class SprintSaleAnnouncementViewModel extends BaseChatViewModel implements Visitable<GroupChatTypeFactory> {

    public static final String SPRINT_SALE_UPCOMING = "flashsale_upcoming";
    public static final String SPRINT_SALE_START = "flashsale_start";
    public static final String SPRINT_SALE_END = "flashsale_end";

    private ArrayList<SprintSaleProductViewModel> listProducts;
    private String redirectUrl;
    private String campaignName;
    private long startDate;
    private long endDate;
    private String sprintSaleType;

    public SprintSaleAnnouncementViewModel(long createdAt, long updatedAt, String messageId,
                                           String senderId, String senderName, String senderIconUrl,
                                           boolean isInfluencer, boolean isAdministrator, String redirectUrl,
                                           ArrayList<SprintSaleProductViewModel> listProducts,
                                           String campaignName, long startDate, long endDate,
                                           String sprintSaleType) {
        super("", createdAt, updatedAt, messageId, senderId, senderName, senderIconUrl,
                isInfluencer, isAdministrator);
        this.redirectUrl = redirectUrl;
        this.listProducts = listProducts;
        this.campaignName = campaignName;
        this.startDate = startDate;
        this.endDate = endDate;
        this.sprintSaleType = sprintSaleType;
    }

    @Override
    public int type(GroupChatTypeFactory typeFactory) {
        return typeFactory.type(this);
    }

    public String getRedirectUrl() {
        return redirectUrl;
    }

    public ArrayList<SprintSaleProductViewModel> getListProducts() {
        return listProducts;
    }

    public String getCampaignName() {
        return campaignName;
    }

    public long getStartDate() {
        return startDate;
    }

    public long getEndDate() {
        return endDate;
    }

    public String getSprintSaleType() {
        return sprintSaleType;
    }
}

File: features/user/groupchat/src/main/java/com/tokopedia/groupchat/chatroom/view/viewmodel/chatroom/SprintSaleAnnouncementViewModel.kt
package com.tokopedia.groupchat.chatroom.view.viewmodel.chatroom

import com.tokopedia.abstraction.base.view.adapter.Visitable
import com.tokopedia.groupchat.chatroom.view.adapter.chatroom.typefactory.GroupChatTypeFactory
import java.util.*

/**
 * @author by nisie on 3/22/18.
 */

class SprintSaleAnnouncementViewModel(val campaignId: String, createdAt: Long, updatedAt: Long,
                                      messageId: String,
                                      senderId: String, senderName: String, senderIconUrl: String,
                                      isInfluencer: Boolean, isAdministrator: Boolean, var redirectUrl: String?,
                                      val listProducts: ArrayList<SprintSaleProductViewModel>,
                                      val campaignName: String, val startDate: Long, val endDate: Long,
                                      val sprintSaleType: String) : BaseChatViewModel("", createdAt, updatedAt, messageId, senderId, senderName, senderIconUrl, isInfluencer, isAdministrator), Visitable<GroupChatTypeFactory> {

    override fun type(typeFactory: GroupChatTypeFactory): Int {
        return typeFactory.type(this)
    }

    companion object {

        const val SPRINT_SALE_UPCOMING = "flashsale_upcoming"
        const val SPRINT_SALE_START = "flashsale_start"
        const val SPRINT_SALE_FINISH = "flashsale_end"
    }
}